### PR TITLE
feat: add `@discordjs/formatters`

### PR DESCRIPTION
**Pull request by suneettipirneni**
**Please describe the changes this PR makes and why it should be merged:**

This PR simply moves formatters out of builders into it's own package. This allows users who use formatters to not have to install the entirely of `/builders`.

**Status and versioning classification:**

- Code changes have been tested against the Discord API, or there are no code changes
- I know how to update typings and have done so, or typings don't need updating
- This PR changes the library's interface (methods or parameters added)
- This PR includes breaking changes (methods removed or renamed, parameters moved or removed)

### CODE REVIEW

**Review by SpaceEEC — APPROVED** [review]

File: .github/labeler.yml [hunk 1]
```diff
@@ -44,3 +43,6 @@ packages:voice:
 packages:ws:
   - packages/ws/*
   - packages/ws/**/*
+packages/formatters:
+  - packages:formatters/*
+  - packages:formatters/**/*
```
Comment (Jiralite) [inline .github/labeler.yml:48]:
```suggestion
packages:formatters:
  - packages/formatters/*
  - packages/formatters/**/*
```

**Review by Jiralite — CHANGES_REQUESTED** [review]

File: packages/formatters/README.md [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+## Contributing
+
+Before creating an issue, please ensure that it hasn't already been reported/suggested, and double-check the
+[documentation][documentation].  
+See [the contribution guide][contributing] if you'd like to submit a PR.
+
+## Help
+
+If you don't understand something in the documentation, you are experiencing problems, or you just need a gentle
+nudge in the right direction, please don't hesitate to join our official [discord.js Server][discord].
```
Comment (Jiralite) [inline packages/formatters/README.md:68]:
```suggestion
If you don't understand something in the documentation, you are experiencing problems, or you just need a gentle nudge in the right direction, please don't hesitate to join our official [discord.js Server][discord].
```

**Review by Jiralite — APPROVED** [review]

File: packages/formatters/LICENSE [hunk 3]
```diff
@@ -0,0 +1,188 @@
...
+      or other liability obligations and/or rights consistent with this
+      License. However, in accepting such obligations, You may act only
+      on Your own behalf and on Your sole responsibility, not on behalf
+      of any other Contributor, and only if You agree to indemnify,
+      defend, and hold each Contributor harmless for any liability
+      incurred by, or claims asserted against, such Contributor by reason
+      of your accepting any such warranty or additional liability.
+
+   END OF TERMS AND CONDITIONS
+
```
Comment (kyranet) [inline packages/formatters/LICENSE:177]: Missing those lines:

https://github.com/discordjs/discord.js/blob/f13ff5c6d35d778c372f7fbb60d3fd947e594404/packages/builders/LICENSE#L178-L179

```suggestion

   Copyright 2021 Noel Buechler
   Copyright 2021 Vlad Frangu

```

File: packages/formatters/package.json [hunk 4]
```diff
@@ -0,0 +1,69 @@
+{
+	"name": "@discordjs/formatters",
+	"version": "1.0.0",
```
Comment (kyranet) [inline packages/formatters/package.json:3]: Are we releasing as v1 instead of v0.1?

File: packages/builders/src/index.ts [hunk 5]
```diff
@@ -1,6 +1,7 @@
 export * as EmbedAssertions from './messages/embed/Assertions.js';
 export * from './messages/embed/Embed.js';
-export * from './messages/formatters.js';
+// TODO: Consider removing this dep in the next major version
+export * from '@discordjs/formatters';
```
Comment (kyranet) [inline packages/builders/src/index.ts:4]: We can make a follow-up PR once this one merges, to remove it, so we can include that change once we start preparing v15.

**Review by kyranet — CHANGES_REQUESTED** [review]
<details>
<summary>The evergrowing list of packages...</summary>

image
<sup>*List goes brrr*</sup>

</details>

I have been considering a couple of things after this PR:

1. Replace some of the overloads with dynamic returns.
2. Moving sanitizers/cleaners (from mainlib) into this package.
3. Maybe sanitize all input before formatting. (@favna and I once discussed removing formatters from Sapphire, but he told me he won't unless Discord.js's formatters sanitize their input like Sapphire does).

**Pull request merged**